resources/regexes.ts

```diff
@@ -695,7 +720,7 @@ export default class Regexes {
     return anyOfArray(array);
   }
 
-  static parse(regexpString: RegExp | string): RegExp {
+  static parse<T extends ParamsAllTypes>(regexpString: RegExp | string): Regex<ParamsAll[T]> {
```

Comment:
So, `Regexes.parse` has been the base function that many things have called for a long time.  See [this code](https://github.com/quisquous/cactbot/blob/0109118d9bbfbfd9abd5a4cca29a2abda28b2136/ui/oopsyraidsy/oopsyraidsy.js#L1217) for example.

One option could be to change these to parse functions to `private parseInternal` or something and have a separate `Regxes.parse` that returns a `RegExp` and then fix up the other callsites later as needed.

(This function returning a typed regex is a great idea!)

Reply:
I assume this is in relation to the other comment about argument types (`regexpString: RegExp | string`)?

I could make a pass through the code and clean them all up but it really seems like it doesn't fit with this PR. Maybe open an issue so it doesn't get lost?

Reply:
Ah no, this is separate from the above comment.  (And the above comment about `(regexpString: RegExp | string) => RegExp` was not about argument types but about the body of the function being duplicated between `NetRegexes.parse` and `Regexes.parse`.)

This comment is about the fact that you are taking a common function called by both Regexes and NetRegexes (i.e. `Regexes.parse`) and making it typed to only return a `Regex<>` but without changing any of the callers that are now using it incorrectly.